### PR TITLE
Stripped down to config file

### PR DESCRIPTION
**Pull request by sphinxc0re**

### CODE REVIEW

File: src/cli/mod.rs [hunk 1]
```diff
@@ -62,67 +77,102 @@ pub fn package() -> DefaultResult<()> {
...
-    Ok(())
+    Ok(Zome {
+        name: zome_config.name.clone(),
+        description: zome_config.description.clone().unwrap_or_default(),
+        config: zome_config.config.clone(),
+        entry_types: Vec::new(),
+        capabilities: caps
+    })
+
+    // let entry_types_dir_path = path.as_ref().join(ENTRY_TYPES_DIR);
```
Comment (thedavidmeister) [inline src/cli/mod.rs]: i don't really like commented out code in develop/master branches :(

File: src/package.rs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        Package {
+            authors,
+            spec_version: (2, 0, 0).into(),
+            dht,
+            properties,
+            dna: Default::default(),
+        }
+    }
+
+    pub fn save_as<T: AsRef<Path>>(&self, path: T) {
```
Comment (thedavidmeister) [inline src/package.rs]: test for this?

File: src/package.rs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+pub struct Package {
+    pub spec_version: Version,
+    pub authors: Vec<Author>,
+    pub dna: Dna,
+    pub properties: Value,
+    pub dht: Dht,
+}
+
+impl Package {
+    pub fn from_app_config(base: AppConfig) -> Package {
```
Comment (thedavidmeister) [inline src/package.rs]: test for this?

File: src/config_files/capability.rs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+impl Capability {
+    pub fn from_file<T: AsRef<Path>>(path: T) -> DefaultResult<Capability> {
+        let file = File::open(&path)?;
+
+        let zome = serde_json::from_reader(file)?;
+
+        Ok(zome)
+    }
+
+    pub fn save_as<T: AsRef<Path>>(&self, path: T) -> DefaultResult<()> {
```
Comment (thedavidmeister) [inline src/config_files/capability.rs]: test?

File: src/config_files/capability.rs [hunk 5]
```diff
@@ -0,0 +1,24 @@
+use error::DefaultResult;
+use serde_json;
+use std::{fs::File, path::Path};
+
+#[derive(Serialize, Deserialize)]
+pub struct Capability {}
+
+impl Capability {
+    pub fn from_file<T: AsRef<Path>>(path: T) -> DefaultResult<Capability> {
```
Comment (thedavidmeister) [inline src/config_files/capability.rs]: test?

**Comment by zippy** [comment]
I think we should merge this branch.  It's ready for that but I do see some things to change, which I'll add as tickets.

**Comment by lucksus** [comment]
I've tried `hcdev package` on https://github.com/holochain/app-spec-rust/pull/1 which pointed out a comma error in a json file. After fixing that it ran through and produced bundle.json even from my manifest.json files. Awesome!

But, it included the rust source files as they were as source (base64 encoded).
I already changed one file into a whole rust project that I could build manually (https://github.com/holochain/app-spec-rust/pull/1/files#diff-913fc95435df0fb77ab7dc294287c3d0). All those files were included as well which made the output quite large.

So right now it does not compile zome source, or I don't know how to put the source code for it to be processed correctly.

Not sure if we see this branch representing ticket #59 or not.

**Comment by sphinxc0re** [comment]
@lucksus If you put a wasm file in a `code/` directory and call the file `code` as well, the resulting dna should resemble a real dna. Source compilation isn't implemented yet, because of the fallback to the generic bundler. The bundler has to detect capabilities at some point which could be done by a special file. This special file could also include build instructions as we discussed on monday

**Pull request merged**